src/screens/TopicPageScreen/MemoizedListComponent.js

```diff
@@ -171,35 +240,30 @@ const RenderListFeed = (props) => {
             showScoreButton={showScoreButton}
           />
         </View>
-        {getCommentLength(item.latest_reactions.comment) > 0 ? (
-          <View style={styles.contentReaction(getHeightReaction())}>
-            <PreviewComment
-              user={item.latest_reactions.comment[0].user}
-              comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
-              image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
-              time={item.latest_reactions.comment[0].created_at}
-              totalComment={getCommentLength(item.latest_reactions.comment) - 1}
-              onPress={onPressComment}
-              item={item.latest_reactions.comment[0]}
-            />
-            <Gap height={8} />
-          </View>
-        ) : (
-          <TouchableOpacity
-            onPress={() => onPressComment(isHaveSeeMore)}
-            style={styles.contentReaction(getHeightReaction())}>
-            <WriteComment
-              postId={''}
-              username={handleUserName(item)}
-              value={''}
-              onChangeText={() => {}}
-              onPress={() => {}}
-              loadingPost={false}
-              isViewOnly={true}
-              withAnonymityLabel={false}
-            />
-          </TouchableOpacity>
-        )}
+        <BlurredLayer
+          layerOnly
+          blurType="light"
+          withToast={true}
+          isVisible={isBlurred}
+          containerStyle={{
+            height: commentHeight()
+          }}>
+          {hasComment ? (
+            <View style={styles.contentReaction(getHeightReaction())}>
+              <PreviewComment
+                user={item.latest_reactions.comment[0].user}
+                comment={item?.latest_reactions?.comment[0]?.data?.text || ''}
+                image={item?.latest_reactions?.comment[0]?.user?.data?.profile_pic_url || ''}
+                time={item.latest_reactions.comment[0].created_at}
+                totalComment={getTotalReaction(item) - 1}
+                item={item.latest_reactions.comment[0]}
+                onPress={onPressComment}
+              />
+            </View>
+          ) : (
+            renderWriteComment()
+          )}
+        </BlurredLayer>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The implementation of the `BlurredLayer` component looks good. However, similar to the previous comment, the `PreviewComment` component inside the `BlurredLayer` seems to be missing some necessary props. The `totalComment` prop is currently calculated as `getTotalReaction(item) - 1`, which may not accurately represent the total number of comments if `getTotalReaction` includes reactions other than comments. Consider using a more accurate method to calculate the total number of comments.
```diff
-                totalComment={getTotalReaction(item) - 1}
+                totalComment={getCountCommentWithChild(item)}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->